google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -109,9 +109,12 @@
    * @return The instance for chaining.
    */
   @Nonnull
-  public T create(
-      @Nonnull DocumentReference documentReference, @Nonnull Map<String, Object> fields) {
-    return performCreate(documentReference, fields);
+  public T create(@Nonnull DocumentReference documentReference, @Nonnull Object fields) {
+    Object data = CustomClassMapper.convertToPlainJavaTypes(fields);
+    if (!(data instanceof Map)) {
+      FirestoreException.invalidState("Can't set a document's data to an array or primitive");
```

Comment:
I have a doubt here, Now we are to support Map and object then should this check be updated?